cmd/podman/system/service.go

```diff
@@ -71,7 +72,21 @@ func service(cmd *cobra.Command, args []string) error {
 
 	logrus.Warn("This function is EXPERIMENTAL")
 	fmt.Fprintf(os.Stderr, "This function is EXPERIMENTAL.\n")
-	return registry.ContainerEngine().RestService(registry.GetContext(), opts)
+
+	// Note that the `system service` command is a bit of an outlier. To
+	// avoid circular dependencies and keep the code clean, `entities` is
+	// not allowed to import `pkg/api` and any sub-packages.  To continue
+	// allowing for a dynamic dispatch and support a parallel ABI & tunnel
+	// support in a single binary, we're doing a dispatch here.
+	podConf := registry.PodmanConfig()
+	switch podConf.EngineMode {
+	case entities.TunnelMode:
```

Comment:
FYI -- `Mode:    []entities.EngineMode{entities.ABIMode}` when registering command implies this case will/should never be called.  If you get here "bad" things are happening.

Reply:
Great catch, thanks!